spec/lib/pantry/helpers/request_verifier_spec.rb

```diff
@@ -0,0 +1,169 @@
+
+    context "Gets a request that does not originate from Alexa" do
+
+      let(:body) do
+        " {\"session\":" +
+          "{\"sessionId\":\"SessionId.b4470d1c-67a0-4da2-98c7-853acead6773\",\"application\":{\"applicationId\":\"amzn1.ask.skill.057ceea6-1e3a-4edb-9a4f-f962f8cc0cac\"},\"attributes\":{},\"user\":{\"userId\":\"amzn1.ask.account.AEP3BKUW344CTHKJDR6P33Y2RJZL6AOSIIWPAEE23O6AIUHMLQ7VSG7NE2CHPDDDCMKF57M43ULCPZLG2U2CIN5NG7MWT6KCDVQS2PXDM5HQYN5J7A26JFPOT7E56W5NQ6S3GSDNINQQZEL5EHDFSXFBM3ST3M5CMK53IZHHQMBCYDA2EFD6OJL7HE6I7WGLSHSLMTN5I7YQDVQ\",\"accessToken\":null},\"new\":false}," +
+          "\"request\":" +
+          "{\"requestId\":\"EdwRequestId.9eaad9b4-48f7-4f6e-8ee9-42a4d7691d4e\",\"locale\":\"en-US\",\"type\":\"IntentRequest\",\"timestamp\":#{Time.now.to_i},\"intent\":{\"name\":\"Warning\",\"slots\":{}}},\"context\":{\"System\": {\"application\": {\"applicationId\": \"applicationId\"}, \"user\":{\"userId\": \"userId\",\"permissions\": {\"consentToken\": \"consentToken\"},\"accessToken\": \"accessToken\"},\"device\": { \"deviceId\": \"deviceId\",\"supportedInterfaces\":{} },\"apiEndpoint\":  \"apiEndpoint\"}}}"
+      end
+
+      context "With an incorrect SignatureCertChainUrl" do
+
+        let(:url) do
+          "https://s3.fakeamazonaws.com/echo.api/echo-api-cert-4.pem"
+        end
+
+        let(:public_key) do
+          double('PublicKey', verify: true)
+        end
+
+        before do
+          allow(Net::HTTP).to receive(:new).and_return(http)
+          allow(OpenSSL::X509::Certificate).to receive(:new).and_return(certificate)
+        end
+
+        it "returns false for a correctly formatted Alexa Request" do
+          request = OpenStruct.new(
+            body: StringIO.new(body),
+            headers: {
+              'SignatureCertChainUrl' => url,
+              'Signature' => signature
+            }
+          )
+
+          verifier = Pantry::Alexa::RequestVerifier.new(request)
+          expect{verifier.verify}.to raise_error Pantry::Alexa::VerificationError
+        end
+
+      end
+
+      context "With a proper SignatureCertChainUrl" do
+
+        let(:url) do
+          "https://s3.amazonaws.com/echo.api/echo-api-cert-4.pem"
+        end
+
+        let(:public_key) do
+          double('PublicKey', verify: false)
+        end
+
+        before do
+          allow(Net::HTTP).to receive(:new).and_return(http)
+          allow(OpenSSL::X509::Certificate).to receive(:new).and_return(certificate)
+        end
+
+        it "returns false for a correctly formatted Alexa Request" do
+          request = OpenStruct.new(
+            body: StringIO.new(body),
+            headers: {
+              'SignatureCertChainUrl' => url,
+              'Signature' => signature
+            }
+          )
+
+          verifier = Pantry::Alexa::RequestVerifier.new(request)
+          expect{verifier.verify}.to raise_error Pantry::Alexa::VerificationError
+        end
+
+      end
+
+    end
+
+  end
+
+end
```

Comment:
Probably scan this whole file for changes to the `let(:body)` and `request = OpenStruct` statements.  I didn't comment on all of them.